newa/cli.py

```diff
@@ -194,62 +193,150 @@ def cmd_event(ctx: CLIContext, errata_ids: list[str]) -> None:
+            # actions already and parent was not found, we abort.
+            if action.parent_id and action.parent_id not in processed_actions:
+                queue_length = len(issue_actions)
+                last_queue_length = endless_loop_check.get(action.id, 0)
+                if last_queue_length == queue_length:
+                    raise Exception(f"Parent {action.parent_id} for {action.id} not found!")
+
+                endless_loop_check[action.id] = queue_length
+                print(f"  skipped for now (parent {action.parent_id} not yet found)")
                 print()
 
                 issue_actions.append(action)
                 continue
 
-            print(f'     Issue would be assigned to {action.assignee}.')
-            print(f'       rendered: >>{render_template(action.assignee, ERRATUM=erratum_job)}<<')
-            print(f'     Will remember the issue as `{action.id}`.')
-            if action.parent_id:
-                print(f'     Issue would have issue `{action.parent_id}` as its parent.')
-            print()
-
-            known_issues[action.id] = True
-
-            # create a fake Issue object for now
-            issue = Issue(id=f'NEWA-{next(jira_id_gen)}')
+            # Find existing issues in Jira related to erratum_job and action.id.
+            #
+            # We re looking for Jira issues such that:
+            #
+            # 1) summary matches rendered_summary (e.g. Errata filelist check) and
+            # 2) description contains erratum_job.erratum (e.g. Erratum(130704) and
+            # 3) description contains erratum_job.erratum.release (e.g. Release(RHEL-8.8.0.Z.EUS))
+            query = f""" summary ~ '{render_template(action.summary, ERRATUM=erratum_job)}' and
+                         description ~ '{erratum_job.erratum!r}' and
+                         description ~ '{erratum_job.erratum.release!r}' """
+
+            # Issues related to the curent respin (a.k.a. new issues).
+            new_issues: list[str] = []
+
+            # Issues related to the previous respin (a.ka. old issues).
+            old_issues: list[str] = []
+
+            for jira_issue in jira.search_open_issue(query):
+                desc = jira_issue.fields.description
+
+                # We found relevant issue in Jira, it is either related to the
+                # current respin (new) or not (old).
+                #
+                # 1) If it is, we won't create it again.
+                if isinstance(desc, str) and desc.find(repr(erratum_job.erratum.builds)):
+                    new_issues.append(jira_issue.key)
+
+                # 2) If it is not, then it is related to the previous respin(s) and need
+                # to be stored for further processing later.
+                else:
+                    old_issues.append(jira_issue.key)
+
+            # We might need to re-use the old issue...
+            if old_issues and action.on_respin == OnRespinAction.KEEP:
```

Comment:
I am no sure this is correct. We should probably clarify the exactl semantics of `OnRespinAction.KEEP` and I would in fact propose to introduce `OnRespinAction.REUSE`, meaning:

- KEEP = do not close but leave it as it is, maybe just update it with a comment about a new issue
- REUSE = do not close but re-use it as a new issue

Either way, old issue(s) should be "reused" only when no new issue have been found. If we already have new issue for that particular jira config item, then we should simply use it.

Reply:
This should be already handled I believe, when there are some KEEP old issues, they are added to new issues list (and removed from old issues list). So when there is already something in new issues list we will end up with multiple issues in that list and that raises an Exception (because of new issues we only allow nothing there or exactly one issue).